Submitting Project 1 (Scissors Paper Stone)

Weimankow opened the pull request.

Weimankow committed:
initial commit

Weimankow committed:
Loop exercise

Weimankow committed:
Day 5 - In class - Madlibs exercise

Weimankow committed:
Project 2: Beat that - 1 player can roll 2 dice

Weimankow committed:
Project 2: Beat it - 1 player can roll and choose their number

Weimankow committed:
Project 2 : Beat it - Second player can roll and choose number

Weimankow committed:
Project 2: Beat it - Win lose draw condition is added

Weimankow committed:
Project 2: Beat it - Base level complete!

Weimankow committed:
Project 2 : Beat that! UI updates

Weimankow committed:
Project 2 - Beat it - added a variable to keep track of each player's wins

DominiqueYeo commented on project1-sps/script.js:
```diff
@@ -1,4 +1,67 @@
 var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+  var computerChoice = Math.ceil(Math.random() * 3);
+  console.log(computerChoice);
+  if (computerChoice === 1) {
+    computerChoice = "scissors";
+  }
+  if (computerChoice === 2) {
+    computerChoice = "paper";
+  }
+  if (computerChoice === 3) {
+    computerChoice = "stone";
+  }
```
This can be written as a function to return you 'scissors' , ' paper' or 'stone'. Looks like a repeatable function that you can abstract out of the main function

DominiqueYeo commented on project1-sps/script.js:
```diff
@@ -1,4 +1,67 @@
+  if (computerChoice === 3) {
+    computerChoice = "stone";
+  }
+  console.log(computerChoice);
```
console.logs should be removed for code reviews for cleaniness

DominiqueYeo commented on project1-sps/script.js:
```diff
@@ -1,4 +1,67 @@
+  }
+  console.log(computerChoice);
+
+  if (
+    (input === "scissors" && computerChoice === "paper") ||
+    (input === "paper" && computerChoice === "stone") ||
+    (input === "stone" && computerChoice === "scissors")
+  ) {
+    playerWins = playerWins + 1;
+    return `You win! <br>
+    You played ${input} and the computer played ${computerChoice}. <br>
+    Score: Player ${playerWins} v.s Computer ${computerWins}. There are ${numDraws} draws. <br>
+    Type 'scissors', 'paper', or 'stone to play another round`;
+  }
+  if (input === computerChoice) {
+    numDraws = numDraws + 1;
+    return `It's a draw! <br>
+    You played ${input} and the computer played ${computerChoice}. <br>
+    Score: Player ${playerWins} v.s Computer ${computerWins}. There are ${numDraws} draws. <br>
+    Type 'scissors', 'paper', or 'stone to play another round`;
+  }
```
good use of if and early returns to control the logic flow of your code

DominiqueYeo replied:
however, it looks like the output  is repeated 3 times, you can create a function to return you your correct output instead of writing it 3 times

DominiqueYeo commented on project1-sps/script.js:
```diff
@@ -1,4 +1,67 @@
+// Computer keeps track of number of times computer has won
+// message shows that
+
+var computerWins = 0;
+var playerWins = 0;
+var numDraws = 0;
```
Good formatting to put global variables together at the top for ease of reading. Variables are also clearly named

DominiqueYeo reviewed (commented):
good effort for your first project. Code is generally clean but look in to using helper functions to help abstract code away from the main function to help with readability

Weimankow committed:
Project 2: Beat That - Added scoring and leaderboard
Score
Keep score for each player.
Leaderboard
When outputting game results in the output box, also output a leaderboard that lists the 2 players and their scores in decreasing order.

Weimankow committed:
Project 2: Beat That - Minor bug fix
When player 2 wins, added 1 to NumofPlayer2Wins

Weimankow committed:
Project 2: Beat That - Refactored leaderboard into a variable

Weimankow committed:
Project 2: Beat That - Minor update to refactored Leaderboard variable

Weimankow committed:
Project 2: Beat That - Added error validation

Weimankow committed:
Project 2: Beat That - Created frontend for third version of game

Weimankow committed:
Project 2: Beat it - Toggled button text from Roll Dice to Submit

Weimankow committed:
Project 2: Beat that - text box can toggle on and off, but buggy- toggle off at wrong moment :(

Weimankow committed:
Project 2: Beat it - Moved leaderboard code from html to js file
